tests/library/debug-controller.spec.ts

```diff
@@ -253,3 +253,22 @@ test('should reset routes before reuse', async ({ server, connectedBrowserFactor
   await expect(page2).toHaveTitle('console.log test');
   await browser2.close();
 });
+
+test('should highlight inside iframe', async ({ backend, connectedBrowser }, testInfo) => {
+  testInfo.annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/33146' });
+
+  const context = await connectedBrowser._newContextForReuse();
+  const page = await context.newPage();
+  await backend.navigate({ url: `data:text/html,<iframe srcdoc="<div>bar</div>"/>` });
+
+
+  await page.frameLocator('iframe').getByText('bar').highlight();
+
+  const highlight = page.frameLocator('iframe').locator('x-pw-highlight');
+  await expect(highlight).not.toHaveCount(0);
+  await backend.hideHighlight();
+  await expect(highlight).toHaveCount(0);
+
+  await backend.highlight({ selector: `frameLocator('iframe').getByText('bar')` });
```

Comment:
Let's also check that `frameLocator().frameLocator().getByText()` does not highlight anything.

Reply:
will do 👍